File: src/core/Account/IAccountStorage.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using core.Account.Responses;

namespace core.Account
{
    public interface IAccountStorage
    {
        Task<AccountStatusView> ViewModel(Guid userId);
        Task SaveViewModel(AccountStatusView user, Guid userId);

        Task<User> GetUserByEmail(string emailAddress);
        Task<User> GetUser(Guid userId);
        Task Save(User u);
        Task Delete(User u);
        Task SaveUserAssociation(ProcessIdToUserAssociation r);
        Task<ProcessIdToUserAssociation> GetUserAssociation(Guid guid);
        Task<IEnumerable<(string email, string id)>> GetUserEmailIdPairs();
    }
}

File: src/core.fs/Shared/Adapters/Storage/IAccountStorage.fs
namespace core.fs.Shared.Adapters.Storage

open System
open System.Collections.Generic
open System.Threading.Tasks
open core.Account
open core.fs.Shared.Domain.Accounts

type IAccountStorage =
    
    abstract member GetUserByEmail: emailAddress:string -> Task<User>
    abstract member GetUser: userId:Guid -> Task<User>
    abstract member Save: u:User -> Task
    abstract member Delete: u:User -> Task
    abstract member SaveUserAssociation: r:ProcessIdToUserAssociation -> Task
    abstract member GetUserAssociation: guid:Guid -> Task<ProcessIdToUserAssociation option>
    abstract member GetUserEmailIdPairs: unit -> Task<IEnumerable<EmailIdPair>>